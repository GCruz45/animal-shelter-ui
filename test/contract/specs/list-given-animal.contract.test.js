import { provider } from "../config/init-pact";
import { Matchers } from "@pact-foundation/pact";
import { AnimalController } from "../../../controllers";

describe("Animal Service", () => {
  describe("When a request to list an animal with name = Bigotes is made", () => {
    beforeAll(async () => {
      await provider.setup();
      await provider.addInteraction({
        uponReceiving: "a request to list animal with name = Bigotes",
        state: "has animal with name = Bigotes",
        withRequest: {
          method: "GET",
          path: "/animals/bigotes",
        },
        willRespondWith: {
          status: 200,
          body: Matchers.eachLike({
            name: "Bigotes",
            breed: Matchers.like("Bengali"),
            gender: Matchers.like("Female"),
            vaccinated: Matchers.boolean(true),
          }),
        },
      });
    });
    test("should return the correct data", async () => {
      const response = await AnimalController.getAnimal("bigotes");
      expect(response.data).toMatchSnapshot();
      await provider.verify();
    });
    afterAll(() => provider.finalize());
  });
});
