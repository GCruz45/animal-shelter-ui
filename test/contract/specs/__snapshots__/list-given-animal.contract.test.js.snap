// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to list an animal with name = Bigotes is made should return the correct data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "name": "Bigotes",
    "vaccinated": true,
  },
]
`;
